src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -504,6 +504,48 @@ public bool IsIronPythonDialogDisabled
             }
         }
 
+        private ICSharpCode.AvalonEdit.TextEditorOptions editTextOptions = new ICSharpCode.AvalonEdit.TextEditorOptions();
+
+        /// <summary>
+        /// Gets the text editor options for python script editor.
+        /// </summary>
+        public ICSharpCode.AvalonEdit.TextEditorOptions TextOptions
```

Comment:
We might want to avoid making this public, just in case at some point we want to change the editor control. Otherwise we would be in a situation similar to the one we had with the Helix upgrade, although with a lot less exposure.

Reply:
this is a valid concern, please do not expose types from our dependencies into our public apis. This was already done for the CodeCompletionProvider and it took significant effort and now technical debt to work around.

Reply:
How do I make `TextOptions ` private and assign its value in viewmodel?
`TextOptions` is needed because `ShowSpaces` and `ShowTabs` are its child members, we cannot access them directly, and we need a binding to this property in order to reflect the changes in real-time.
On the other hand `ShowTabsAndSpacesInScriptEditor` is needed because we have a global setting option that is binded to its bool value.

Reply:
You don't need to use this type directly in this property, you can use an interface or a class that hides the underlying type.